### PR TITLE
0D test for split layer

### PR DESCRIPTION
**Pull request by Abdurrahheem**
This PR introduces parametrized `0/1D` input support test for `Split` layer.

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [x] There is a reference to the original bug report and related work
- [x] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [x] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

File: modules/dnn/test/test_layers_1d.cpp
Comment (dkurt) [inline modules/dnn/test/test_layers_1d.cpp:278]: I think it's critical here to add 1D but with multiple values here (see https://github.com/opencv/opencv/pull/25074#discussion_r1522654268). Even with `{2}` to check that Split will really split 1D vector to two parts.

**Review by dkurt — CHANGES_REQUESTED** [review]

File: modules/dnn/test/test_layers_1d.cpp
Comment (Abdurrahheem) [reply modules/dnn/test/test_layers_1d.cpp:278]: `Split` later just copies one tensor into multiple([see](https://github.com/opencv/opencv/blob/60ca3d259eff77bf6287c07bc8db66078d5c6cd3/modules/dnn/src/layers/split_layer.cpp#L94C5-L105C7)). The one you mention is a `slice`.

Comment (dkurt) [reply modules/dnn/test/test_layers_1d.cpp]: Got it, thanks. Very strange layer, to be honest :) But let's keep test enabled

**Review by dkurt — APPROVED** [review]

**Comment by Abdurrahheem** [comment]
@dkurt do you have PR merge rights?

**Comment by dkurt** [comment]
@Abdurrahheem, I don't have

**Comment by Abdurrahheem** [comment]
@vpisarev could you please merge this PR

File: modules/dnn/test/test_layers_1d.cpp
Comment (opencv-alalek) [inline modules/dnn/test/test_layers_1d.cpp:282]: @vpisarev What is wrong with Mat API if we need to do that?

Comment (Abdurrahheem) [reply modules/dnn/test/test_layers_1d.cpp]: No thing is wrong. I have changed initialization scheme.

**Comment by vpisarev** [comment]
1. please rename test_layers_1d.cpp => test_layers_nd.cpp
2. use empty shape vector in 0D case

File: modules/dnn/test/test_layers_1d.cpp
Comment (opencv-alalek) [inline modules/dnn/test/test_layers_1d.cpp:280]: make sense to add `{4, 5}`

Comment (Abdurrahheem) [reply modules/dnn/test/test_layers_1d.cpp]: Done

**Comment by Abdurrahheem** [comment]
> 1. please rename test_layers_1d.cpp => test_layers_nd.cpp
> 2. use empty shape vector in 0D case

Can we rename it after merging the sequence of tests of 0/1D to avoid conflicts?

File: modules/dnn/test/test_layers_1d.cpp
Comment (dkurt) [inline modules/dnn/test/test_layers_1d.cpp:302]: @asmorkalov, this test should fail (https://github.com/opencv/opencv/pull/25231#issuecomment-2015359780)

Comment (asmorkalov) [reply modules/dnn/test/test_layers_1d.cpp:302]: Got it. Let's  discuss it on Friday call. I'll a bit out-of-context now.

**Pull request merged**
